pilot/pkg/model/sidecar_test.go

```diff
@@ -1888,34 +2009,38 @@ func TestCreateSidecarScope(t *testing.T) {
 			}
 
 			for _, s1 := range sidecarScope.services {
-				found = false
+				serviceFound = false
+				portsMatched = false
+				var ports PortList
 				for _, s2 := range tt.excpectedServices {
 					if s1.Hostname == s2.Hostname {
+						serviceFound = true
 						if len(s2.Ports) > 0 {
 							if reflect.DeepEqual(s2.Ports, s1.Ports) {
-								found = true
-								break
+								portsMatched = true
+							} else {
+								ports = s2.Ports
 							}
-						} else {
-							found = true
-							break
 						}
+						break
 					}
 				}
-				if !found {
+				if !serviceFound {
 					t.Errorf("Expected service %v in SidecarScope but not found", s1.Hostname)
+				} else if len(ports) > 0 && !portsMatched {
+					t.Errorf("Expected service %v found in SidecarScope but ports not merged correctly. want: %v, got: %v", s1.Hostname, s1.Ports, ports)
```

Comment:
%v can print ports. And maybe %+v

Reply:
I tried both earlier and both printed this
```
Expected service proxy found in SidecarScope but ports not merged correctly. want: [0x14000f5adb0 0x1075d38e0], got: [0x1075d3820 0x1075d38a0 0x1075d38e0]
```

That is why I added that String() method